### DISCUSSION TITLE
Estou com duvida nesse exercício

### DISCUSSION THREAD

**Discussion by H3l1opy** [community/community#13170] [Projects and Issues]
Eu nao sei no que errei por que quando eu escolho qualquer numero a variavel "z" sempre aparece e quando eu escolho quantas vezes eu vou parcelar no cartao o resto do comando nao aparece 

No comesso so tinha uma condiçao com varios "elif" mais como nao funciono eu tentei dividir em duas condições e nao deu dai eu dividi em 3 condições eu to a algumas horas sem saber oque esta errado ai 

print(' ')
print('=====|DESAFIO 44|=====')
'''
elabore um programa que calcule o 
valor a ser pago por um produto. 
considerando o seu preço normal e 
condição de pagamento:

- a vista dinheiro/cheque: 10% de 
desconto
- a vista no cartão:5% de desconto
- em ate 2x no cartao: preço normal 
- 3x ou mais no cartão: 20% de juros
'''
print(' ')
a = float(input('valor a pagar ? : R$ '))
print(' ')
b = input('''[1] a vista no dinheiro/cheque \n[2] a vista no cartão \n[3] em ate 2x no cartão \n[4] 3x ou mais no cartão 
\nqual meio de pagamento ? : ''' )
c = a * (10/100)
d = a * (5/100)
e = ((20/100) * a) / 3
ea = ((20/100) * a) / 4
eb = ((20/100) * a) / 5
ec = ((20/100) * a) / 6
f = a / 2
z = input('em quantas vezes ? : ' )
print(' ')
#print('{}'format(b))
if b == 1:
    print('o valor a pagar agora e R${} '.format(c))
elif b == 2:
    print('o valor a pagar agora e R${} '.format(d))   
elif b == 3:
    print('o valor a pagar e 2x de R${}  '.format(f))           	
elif b == 4:
    print(f'{z}')
else:
    print(' ')    
          
if z == 3:
    print('o valor a pagar agora e R${} '.format(e))
elif z == 4:
    print('o valor a pagar agora e R${} '.format(ea))    
elif z == 5:
    print('o valor a pagar agora e R${} '.format(eb))        
elif z == 6:
    print('o valor a pagar agora e R${} '.format(ec))   
else:
    print(' ')

**Comment by Shockraa** [comment]
Olá amigo, fiz umas alterações no seu código com a explicação abaixo.

```
print(' ')
a = float(input('valor a pagar ? : R$ '))
print(' ')
b = int(input('''[1] a vista no dinheiro/cheque
[2] a vista no cartão
[3] em ate 2x no cartão
[4] 3x ou mais no cartão
\nqual meio de pagamento ? : '''))
c = a * (10/100)
d = a * (5/100)
f = a / 2


if b == 1:
    print('o valor a pagar agora é R${:.2f}'.format(a - c))
elif b == 2:
    print('o valor a pagar agora é R${:.2f}'.format(a - d))
elif b == 3:
    print('o valor a pagar é 2x de R${:.2f}'.format(f))
elif b == 4:
    z = int(input('em quantas vezes ? : '))
    print(' ')
    if z >= 3:
        parcela = a * 1.2 / z
        print('o valor a pagar agora é R${:.2f}'.format(a * 1.2))
        print('parcelado em {}x de R${:.2f}'.format(z, parcela))
    else:
        print('Quantidade de parcelas inválida. Mínimo de 3 parcelas.')
else:
    print('Forma de pagamento inválida.')
```
    
    Para fazer a variável "z" parar de aparecer sempre, coloquei ela dentro do condicional (elif b == 4) para que ela seja acionada apenas quando o usuário escolher a opção de 3x ou mais no cartão.
    Reduzi as variáveis fazendo o calculo delas dentro das condições.
    Na condição (elif b == 4), ele calcula o 20% de juros exibindo o novo valor e as parcelas.
   
    
    Espero ter ajudado!
